### PR TITLE
Diagnostics

### PR DESCRIPTION
**Pull request by raskle**
## Overview
Diagnostics Client implementation

- [x] I have read the [contributing guide](https://github.com/pilosa/pilosa/blob/master/CONTRIBUTING.md).
- [x] I have agreed to the [Contributor License Agreement](https://cla-assistant.io/pilosa/pilosa).
- [x] I have updated the [documentation](https://github.com/pilosa/pilosa/tree/master/docs).
- [x] I have resolved any merge conflicts.
- [x] I have included tests that cover my changes.
- [x] All new and existing tests pass.

## Code review checklist
This is the checklist that the reviewer will follow while reviewing your pull request. You do not need to do anything with this checklist, but be aware of what the reviewer will be looking for.

- [ ] Ensure that any changes to external docs have been included in this pull request.
- [ ] If the changes require that minor/major versions need to be updated, tag the PR appropriately.
- [ ] Ensure the new code is [properly commented](https://github.com/golang/go/wiki/CodeReviewComments#doc-comments) and follows [Idiomatic Go](https://dmitri.shuralyov.com/idiomatic-go).
- [ ] Check that tests have been written and that they cover the new functionality.
- [ ] Run tests and ensure they pass.
- [ ] Build and run the code, performing any applicable integration testing.

### CODE REVIEW

File: docs/configuration.md [hunk 1]
```diff
@@ -206,6 +206,19 @@ Any flag that has a value that is a comma separated list on the command line bec
...
+##### Metric Diagnostics
+
+* Description: Enable diagnostic reporting. To disable diagnostics set to false.
+* Flag: `metric.diagnostics`
+* Env: `PILOSA_METRIC_DIAGNOSTICS`
+* Config:
+
+    ```toml
+    [metric]
+    diagnostics = true
```
Comment (codysoyland) [inline docs/configuration.md]: Seems that `false` is not a valid value for the falsy value. `0` seems to disable it. Would be good to notify user if the value is not valid.

File: server.go [hunk 2]
```diff
@@ -507,9 +514,66 @@ func (s *Server) checkMaxSlices(scheme string, hostPort string) (map[string]uint
 	return pb.MaxSlices, nil
 }
 
+// monitorDiagnostics periodically polls the the Pilosa Indexes for cluster info.
+func (s *Server) monitorDiagnostics() {
+	if s.DiagnosticInterval <= 0 {
+		s.Logger().Printf("diagnostics disabled")
```
Comment (codysoyland) [inline server.go]: Might be a good idea to write out a message here if diagnostics is enabled, so it's transparent to the user. Maybe a little blurb with a link to the documentation would be nice, ie. "Pilosa is currently configured to send small diagnostics reports every hour to our team. More information here: http...."

File: docs/configuration.md (hunk 1 above)
Comment (raskle) [reply docs/configuration.md]: I tested this and false does seem to work.  I updated to docs with clearer instructions.

File: server.go (hunk 2 above)
Comment (raskle) [reply server.go]: done

Comment (travisturner) [reply server.go]: "the the"

**Pull request merged**